cmake/XrplSanitizers.cmake

```diff
@@ -0,0 +1,152 @@
+elseif(ENABLE_TSAN)
+    list(APPEND SANITIZERS_FLAGS "thread")
+endif()
+
+if(ENABLE_UBSAN)
+    # UB sanitizer flags
+    if(IS_CLANG)
+        # Clang supports additional UB checks
+        list(APPEND SANITIZERS_FLAGS "undefined" "float-divide-by-zero" "unsigned-integer-overflow")
+    else()
+        list(APPEND SANITIZERS_FLAGS "undefined" "float-divide-by-zero")
+    endif()
+endif()
+
+# Configure code model for GCC on amd64
+# Use large code model for ASAN to avoid relocation errors
+# Use medium code model for TSAN (large is not compatible with TSAN)
+set(SANITIZERS_RELOCATION_FLAGS "")
+if(IS_GCC AND IS_AMD64)
+    if(ENABLE_ASAN)
+        message(STATUS "  Using large code model (-mcmodel=large)")
+        list(APPEND SANITIZERS_COMPILE_FLAGS "-mcmodel=large")
+        set(SANITIZERS_RELOCATION_FLAGS "-mcmodel=large")
+    elseif(ENABLE_TSAN)
+        message(STATUS "  Using medium code model (-mcmodel=medium)")
+        list(APPEND SANITIZERS_COMPILE_FLAGS "-mcmodel=medium")
+        set(SANITIZERS_RELOCATION_FLAGS "-mcmodel=medium")
+    endif()
+endif()
+
+# Compiler-specific configuration
+if(IS_GCC)
+    # Disable mold, gold and lld linkers for GCC with sanitizers
+    # Use default linker (bfd/ld) which is more lenient with mixed code models
+    set(use_mold OFF CACHE BOOL "Use mold linker" FORCE)
+    set(use_gold OFF CACHE BOOL "Use gold linker" FORCE)
+    set(use_lld OFF CACHE BOOL "Use lld linker" FORCE)
+    message(STATUS "  Disabled mold, gold, and lld linkers for GCC with sanitizers")
+
+    # Suppress false positive warnings in GCC with stringop-overflow
+    list(APPEND SANITIZERS_COMPILE_FLAGS "-Wno-stringop-overflow")
+
+    if(ENABLE_TSAN)
+        # GCC doesn't support atomic_thread_fence with tsan. Suppress warnings.
+        list(APPEND SANITIZERS_COMPILE_FLAGS "-Wno-tsan")
+    endif()
+
+    # Join sanitizer flags with commas for -fsanitize option
+    list(JOIN SANITIZERS_FLAGS "," SANITIZERS_FLAGS_STR)
+
+    # Add sanitizer to compile and link flags
+    list(APPEND SANITIZERS_COMPILE_FLAGS "-fsanitize=${SANITIZERS_FLAGS_STR}")
+    set(SANITIZERS_LINK_FLAGS "${SANITIZERS_RELOCATION_FLAGS}" "-fsanitize=${SANITIZERS_FLAGS_STR}")
+
+elseif(IS_CLANG)
+    # Add ignorelist for Clang (GCC doesn't support this)
+    # Use CMAKE_SOURCE_DIR to get the path to the ignorelist
+    set(IGNORELIST_PATH "${CMAKE_SOURCE_DIR}/sanitizers/suppressions/sanitizer-ignorelist.txt")
+    if(NOT EXISTS "${IGNORELIST_PATH}")
+        message(FATAL_ERROR "Sanitizer ignorelist not found: ${IGNORELIST_PATH}")
+    endif()
+
+    list(APPEND SANITIZERS_COMPILE_FLAGS "-fsanitize-ignorelist=${IGNORELIST_PATH}")
+    message(STATUS "  Using sanitizer ignorelist: ${IGNORELIST_PATH}")
+
+    # Join sanitizer flags with commas for -fsanitize option
+    list(JOIN SANITIZERS_FLAGS "," SANITIZERS_FLAGS_STR)
+
+    # Add sanitizer to compile and link flags
+    list(APPEND SANITIZERS_COMPILE_FLAGS "-fsanitize=${SANITIZERS_FLAGS_STR}")
```

Comment:
nit: this can be moved out of if, to remove duplication, but it's ok to keep as is

Reply:
Ok